test/Analyzers/Microsoft.Extensions.ExtraAnalyzers/Common/ConditionalAccessAnalyzerTests.cs

```diff
@@ -106,7 +106,9 @@ public Arg(TParameter arg1)
             null,
             new[] { Source }).ConfigureAwait(false);
 
-#if NET6_0_OR_GREATER
+#if NET8_0_OR_GREATER
+        // Whilst these API are marked as NET6_0_OR_GREATER we don't build .NET 6.0,
```

Comment:
I'm not understanding here what you mean here by "marked as NET6.0 or greater"

Reply:
The actual API are guarded with `#if NET6_0_OR_GREATER` meaning those are technically available in .NET 6.

Reply:
But the actual API isn't building for net6.0, so no matter what the ifdef says, the availability of an API is decided by whether or not the API can be called in 6.0, not if the API could theoretically be available but is only compiled for .net 8.

Reply:
Until we removed `#if` defs, it can create a confusion - e.g., a developer looking at the code and seeing an API guarded with `#if NET6_0_OR_GREATER` then running a test under .NET 6 and seeing a failure - I've gone through that experience and it was confusing.
When we remove the pragmas (if ever) then we can remove the comments too.